### PR TITLE
🔧 Fix `bundler` version to `2.4.22`

### PR DESCRIPTION
**Pull request by connormaglynn**
## 👀 Purpose

- In relation to https://github.com/ministryofjustice/operations-engineering/issues/4081
- To fix the Docker image failing at build

## ♻️ What's changed

- Fixed bundler version to `2.4.22` 
- Regenerate `Gemfile.lock` file with the new bundler version
- Added `Build Docker Image` on Pull Request to `main` to ensure changes don't break the image build phase before merging the code into the main branch
- Some workflow refactors in terms of naming

## 📝 Notes

- v2.5 of the bundler dropped support for ruby 2.7 ([REF](https://github.com/rubygems/rubygems/blob/master/bundler/CHANGELOG.md#250-december-15-2023)). So I've fixed the version to the suggested `2.4.22`

### CODE REVIEW

File: .github/workflows/cicd-dependency-review.yml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # Need a GitHub Advanced Security license to run this action on private repos.
 
-name: Dependency Review
+name: ♻️ Dependency Review
 on:
   pull_request:
-    types: [opened, edited, reopened, synchronize]
+    branch: main
```
Comment (tamsinforbes) [inline .github/workflows/cicd-dependency-review.yml:6]: was this run on any branch previously?

File: .github/workflows/format-code.yml
Comment (tamsinforbes) [inline .github/workflows/format-code.yml:1]: 🔥

**Review by tamsinforbes — APPROVED** [review]
🌝🌖🚀LGTM🚀🌘🌚

File: .github/workflows/cicd-dependency-review.yml (hunk 1 above)
Comment (connormaglynn) [reply .github/workflows/cicd-dependency-review.yml:6]: Yes 😁 But the previous syntax is a bit too lax, in the sense that you end up running CICD checks for things like the PR title changing 🏃

**Pull request merged**